pandas/tests/indexing/test_multiindex.py

```diff
@@ -198,6 +202,27 @@ def test_loc_getitem_array(self):
         result = x.loc[scalar]
         tm.assert_series_equal(result, expected)
 
+    def test_loc_generator(self):
```

Comment:
there does not appear to be an equivalent test on Index with a generator, or even an iterator. Sure they *should* work. But is this fixed in this PR? or just adding testing to something already works?